File: Android/2016-04-13-N-Preview-Recycler-Lambda-Example/app/src/main/java/tech/thdev/recycler_lambda_example/MainActivity.java
package tech.thdev.recycler_lambda_example;

import android.net.Uri;
import android.os.Bundle;
import android.support.annotation.DrawableRes;
import android.support.design.widget.BottomSheetBehavior;
import android.support.design.widget.BottomSheetDialog;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuItem;
import android.view.View;
import android.widget.ImageView;
import android.widget.Toast;

import com.google.android.gms.appindexing.Action;
import com.google.android.gms.appindexing.AppIndex;
import com.google.android.gms.appindexing.Thing;
import com.google.android.gms.common.api.GoogleApiClient;

import java.util.ArrayList;

import tech.thdev.recycler_lambda_example.adapter.ImageViewRecyclerAdapter;
import tech.thdev.recycler_lambda_example.presenter.MainPresenter;
import tech.thdev.recycler_lambda_example.presenter.MainPresenterImpl;

public class MainActivity extends AppCompatActivity implements MainPresenter.View {

    private FloatingActionButton fab;
    private MainPresenter mainPresenter;
    /**
     * ATTENTION: This was auto-generated to implement the App Indexing API.
     * See https://g.co/AppIndexing/AndroidStudio for more information.
     */
    private GoogleApiClient client;

//    private BottomSheetBehavior behavior;
//    private View bottomSheet;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);

        mainPresenter = new MainPresenterImpl(this);

//        bottomSheet = findViewById(R.id.rl_bottom_sheet);
//        behavior = BottomSheetBehavior.from(bottomSheet);

        fab = (FloatingActionButton) findViewById(R.id.fab);
        fab.setOnClickListener(view ->
                Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                        .setAction("Action", null).show()
        );

        findViewById(R.id.btn_on_click).setOnClickListener(view -> {
            Snackbar.make(view, "Lambda with your own action", Snackbar.LENGTH_SHORT).show();
            Toast.makeText(this, "aaa", Toast.LENGTH_SHORT).show();
        });

        ArrayList<Image> imageList = new ArrayList<>();
        imageList.add(new Image(R.drawable.sample_image));

        ImageViewRecyclerAdapter adapter = new ImageViewRecyclerAdapter(this);
        adapter.setItemList(imageList);
        adapter.setOnTouchLongClickListener((Image, position) -> {
            mainPresenter.onImageItemLongClick(R.drawable.sample_image);
            return true;
        });

        RecyclerView recyclerView = (RecyclerView) findViewById(R.id.recycler_view);
        recyclerView.setAdapter(adapter);
        // ATTENTION: This was auto-generated to implement the App Indexing API.
        // See https://g.co/AppIndexing/AndroidStudio for more information.
        client = new GoogleApiClient.Builder(this).addApi(AppIndex.API).build();
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        int id = item.getItemId();

        //noinspection SimplifiableIfStatement
        if (id == R.id.action_settings) {
            return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    public void showFlickerImageBottomSheet(@DrawableRes int resource) {
        BottomSheetDialog bottomSheetDialog = new BottomSheetDialog(this);
        View view = getLayoutInflater().inflate(R.layout.bottom_sheet_image_view, null);

        ImageView imageView = (ImageView) view.findViewById(R.id.image);
        imageView.setImageResource(resource);

        bottomSheetDialog.setContentView(view);
        bottomSheetDialog.show();
    }

    /**
     * ATTENTION: This was auto-generated to implement the App Indexing API.
     * See https://g.co/AppIndexing/AndroidStudio for more information.
     */
    public Action getIndexApiAction() {
        Thing object = new Thing.Builder()
                .setName("Main Page") // TODO: Define a title for the content shown.
                // TODO: Make sure this auto-generated URL is correct.
                .setUrl(Uri.parse("http://[ENTER-YOUR-URL-HERE]"))
                .build();
        return new Action.Builder(Action.TYPE_VIEW)
                .setObject(object)
                .setActionStatus(Action.STATUS_TYPE_COMPLETED)
                .build();
    }

    @Override
    public void onStart() {
        super.onStart();

        // ATTENTION: This was auto-generated to implement the App Indexing API.
        // See https://g.co/AppIndexing/AndroidStudio for more information.
        client.connect();
        AppIndex.AppIndexApi.start(client, getIndexApiAction());
    }

    @Override
    public void onStop() {
        super.onStop();

        // ATTENTION: This was auto-generated to implement the App Indexing API.
        // See https://g.co/AppIndexing/AndroidStudio for more information.
        AppIndex.AppIndexApi.end(client, getIndexApiAction());
        client.disconnect();
    }
}

File: app/src/main/java/tech/thdev/app/ui/main/MainActivity.kt
package tech.thdev.app.ui.main

import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import androidx.appcompat.app.AppCompatActivity
import androidx.appcompat.widget.Toolbar
import androidx.fragment.app.Fragment
import com.google.android.material.bottomnavigation.BottomNavigationView
import tech.thdev.app.R
import tech.thdev.app.ui.basic.PlaceholderFragment
import tech.thdev.app.ui.dailyboxoffice.DailyBoxOfficeFragment
import tech.thdev.app.ui.weekboxoffice.WeeklyBoxOfficeFragment

class MainActivity : AppCompatActivity() {

    private val bottomNavigationView: BottomNavigationView by lazy {
        findViewById<BottomNavigationView>(R.id.bottom_navigation)
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val toolbar = findViewById<Toolbar>(R.id.toolbar)
        setSupportActionBar(toolbar)

        initView()
    }

    private val fragmentMap = mutableMapOf<Int, Fragment>()

    private fun initView() {
        fragmentMap[R.id.action_one] = PlaceholderFragment.newInstance()
        fragmentMap[R.id.action_two] = DailyBoxOfficeFragment.newInstance()
        fragmentMap[R.id.action_three] = WeeklyBoxOfficeFragment.newInstance()

        bottomNavigationView.setOnNavigationItemSelectedListener {
            fragmentMap[it.itemId]?.let { fragment ->
                replaceFragment(fragment)
                true
            } ?: false
        }
        bottomNavigationView.selectedItemId = R.id.action_one
    }

    private fun replaceFragment(fragment: Fragment) {
        supportFragmentManager
            .beginTransaction()
            .replace(R.id.container, fragment)
            .commitAllowingStateLoss()
    }

    override fun onCreateOptionsMenu(menu: Menu?): Boolean {
        // Inflate the menu; this adds items to the action bar if it is present.
        menuInflater.inflate(R.menu.menu_main, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        // Handle action bar item clicks here. The action bar will
        // automatically handle clicks on the Home/Up button, so long
        // as you specify a parent activity in AndroidManifest.xml.
        return if (item.itemId == R.id.action_settings) {
            true
        } else {
            super.onOptionsItemSelected(item)
        }
    }
}